<launch>

    <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find wpr_simulation)/worlds/robocup_home.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="recording" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!-- Spawn a robot into Gazebo -->
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-file $(find wpr_simulation)/models/wpb_home.model -urdf -x -0.0 -y -0.0 -model wpb_home" />

    <!--- Run AMCL -->
    <include file="$(find wpb_home_tutorials)/nav_lidar/amcl_omni.launch" />

    <!-- RViz and TF tree -->
    <arg name="model" default="$(find wpb_home_bringup)/urdf/wpb_home.urdf"/>
    <arg name="gui" default="false" />
    <arg name="rvizconfig" default="$(find map_provider)/rviz/rrt.rviz" />

    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
        <rosparam command="load" file="$(find wpb_home_bringup)/config/wpb_home.yaml" />
    </node>
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

    <!--- Run move base -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base"  output="screen">
        <rosparam file="$(find wpb_home_tutorials)/nav_lidar/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find wpb_home_tutorials)/nav_lidar/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find wpb_home_tutorials)/nav_lidar/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find wpb_home_tutorials)/nav_lidar/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find wpb_home_tutorials)/nav_lidar/local_planner_params.yaml" command="load" />
        <param name="base_global_planner" value="global_planner/GlobalPlanner" /> 
        <param name="use_dijkstra" value="true"/>
        <param name="base_local_planner" value="wpbh_local_planner/WpbhLocalPlanner" />
        <param name= "controller_frequency" value="10" type="double"/>
    </node>

</launch>